packages/common/lib/xml-ns-key.js

```diff
@@ -0,0 +1,38 @@
+// The xml parser takes care of validating the attribute name.
+// If the user started the attribute name with "xmlns:" we can assume that
+// they meant for it to be an xml namespace attribute.
+// xmlns attributes explicitly can't contain ":" after the "xmlns:" part.
+const namespaceRegex = /^xmlns(?<prefixWithColon>:(?<prefix>[^:]*))?$/;
+
+function isXMLNamespaceKey(key, includeEmptyPrefix) {
+  if (typeof key !== "string") {
+    return false;
+  }
+  const matchArr = key.match(namespaceRegex);
+  if (matchArr === null) {
+    return false;
+  }
+  if (includeEmptyPrefix === true) {
```

Comment:
`  if (includeEmptyPrefix === true) {` correctness depends on running **after**:
`	  if (matchArr === null) {
    return false;
  }`

should an `else if` be used to explicitly mark this relationship?

Reply:
I prefer not to, would it be enough to add a comment?